src/envoy/http/jwt_auth/jwt_authenticator.cc

```diff
@@ -195,8 +198,10 @@ void JwtAuthenticator::VerifyKey(const PubkeyCacheItem& issuer_item) {
     return;
   }
 
-  // TODO: check forward_payload_header.
-  headers_->addReferenceKey(kJwtPayloadKey, jwt_->PayloadStrBase64Url());
+  LowerCaseString key(issuer_item.jwt_config().forward_payload_header());
+  if (!jwt_->PayloadStrBase64Url().empty()) {
+    headers_->addCopy(key, jwt_->PayloadStrBase64Url());
```

Comment:
You can do this to avoid creating temp variable
headers_->addCopy(LowerCaseString{issuer_item.jwt_config().forward_payload_header()}, jwt_->PayloadStrBase64Url());

Reply:
Also check forward_payload_header.empty()

Reply:
Done.